helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -359,7 +366,17 @@ public void scheduleRebalance(long rebalanceTime) {
    * Schedule an on demand rebalance pipeline.
    * @param delay
    */
+  @Deprecated
   public void scheduleOnDemandRebalance(long delay) {
+    scheduleOnDemandRebalance(delay, true);
+  }
+
+  /**
+   * Schedule an on demand rebalance pipeline.
+   * @param delay
+   * @param shouldRefreshCache true if refresh the cache before scheduling a rebalance.
+   */
+  public void scheduleOnDemandRebalance(long delay, boolean shouldRefreshCache) {
```

Comment:
Please correct me if I misunderstand the boolean value's purpose, I understand it's useful if wagedRebalancer would like to calculate a global rebalance on a "snapshot", "old" version of the cache object. Is it harder to maintain and debug the cache object if we sometimes have an update-to-date cache and sometimes an old version cache? 
Is it possible to take the snapshot of cache object at some time and give it to waged rebalancer for the global calculation?

Reply:
It is in our roadmap. But take a snapshot of the cache is a large project itself. For now, we don't have that feature yet.